webapp/test_results_handler.go

```diff
@@ -131,36 +131,33 @@ func parseTestRunUIFilter(r *http.Request) (filter testRunUIFilter, err error) {
 
 		const singleRunURL = `/api/run?sha=%s&product=%s`
 		var specs []string
-		if beforeDecoded, err := base64.URLEncoding.DecodeString(before); err == nil {
-			var run shared.TestRun
-			if err = json.Unmarshal([]byte(beforeDecoded), &run); err != nil {
-				return filter, err
+		beforeSpec, err := shared.ParseProductSpec(before)
```

Comment:
Why do we treat `before`/`after` two different ways? IIUC, we either:

1. Find a `before`/`after` spec in plaintext and add it to `specs`, OR
1. Find a `before`/`after` spec base64-encoded and copy it to `filter.{Before|After}TestRun`.

In non-error cases we do one of these things (write to `specs` OR write to `filter`), but not both. Why the two different ways of interpreting the same input?

Reply:
Specs: `before=chrome@latest`
Base64: `before=[encodedblobofchars]` where decoding the blob is a JSON object `{browser_name: "foo", ... }`

It's an old use-case for passing in arbitrary runs that aren't in the datastore, while still being able to see them in the UI. This PR doesn't change behaviour, just the order in which we try base64'ing